src/scr.c

```diff
@@ -37,6 +37,19 @@
 #include "dtcmp.h"
 #endif /* HAVE_LIBDTCMP */
 
+/* define which state we're in for API calls, this is to help ensure
+ * users call SCR functions in the correct order */
+typedef enum {
+    SCR_STATE_UNINIT,     /* before init and after finalize */
+    SCR_STATE_INIT,       /* between init/finalize */
```

Comment:
I'm not sure I like the name "INIT" here because the states are then not mutually exclusive. Perhaps "IDLE"? or some other term.